src/main/java/com/endava/internship/warmup/domain/service/ArrayProcessorWithForLoops.java

```diff
@@ -1,153 +1,327 @@
      */
     @Override
-    public boolean allMatch(final String[] input,
-                            ToIntFunction<String> function,
-                            IntPredicate predicate) {
-        throw new UnsupportedOperationException("Method not implemented yet");
+    public boolean allMatch(final String[] input, ToIntFunction<String> function, IntPredicate predicate) {
+        int size = input.length;
+        int count = 0;
+        boolean rt = false;
+        for (String s : input) {
+            if (predicate.test(function.applyAsInt(s))) {
+                count++;
+            }
+        }
+        if (count == size) {
+            rt = true;
+        }
+        return rt;
     }
 
     /**
      * Copy values into a separate array from specific index to stopindex
-     * @param input non-null array of ints
+     *
+     * @param input          non-null array of ints
      * @param startInclusive the first index of the element from input to be included in the new array
-     * @param endExclusive the last index prior to which the elements are to be included in the new array
+     * @param endExclusive   the last index prior to which the elements are to be included in the new array
      * @throws IllegalArgumentException when parameters are outside of input index bounds
      */
     @Override
     public int[] copyValues(int[] input, int startInclusive, int endExclusive) throws IllegalArgumentException {
-        throw new UnsupportedOperationException("Method not implemented yet");
+        if (startInclusive < 0 || startInclusive > endExclusive) {
+            throw new IllegalArgumentException("copy value out of bounds");
+        }
+        int arr[] = new int[endExclusive - startInclusive];
+        int j = 0;
+        for (int i = startInclusive; i < endExclusive; i++) {
+            arr[j] = input[i];
+            j++;
+        }
+        return arr;
     }
 
     /**
      * Replace even index values with their doubles and odd indexed elements with their negative
+     *
      * @param input non-null immutable array of ints
      * @return new array with changed elements
      */
     @Override
     public int[] replace(final int[] input) {
-        throw new UnsupportedOperationException("Method not implemented yet");
+        int[] arr = new int[input.length];
+        for (int i = 0; i < input.length; i++) {
+            if (input[i] % 2 == 0) {
+                arr[i] = input[i] * 2;
+            } else if (input[i] % 2 != 0) {
+                arr[i] = -input[i];
+            }
+        }
+
+        return arr;
     }
 
     /**
      * Find the second max value in the array
+     *
      * @param input non-null immutable array of ints
      */
     @Override
     public int findSecondMax(final int[] input) {
-        throw new UnsupportedOperationException("Method not implemented yet");
+        int[] fsmArr = new int[input.length];
```

Comment:
As I see, this implementation only removes duplicates from array and returns penultimate element from array
But task requires to `Find the second max value in the array`. Consider to rewrite

Reply:
Done